channels/layers.py

```diff
@@ -279,19 +282,18 @@ def _clean_expired(self):
                 self._remove_from_groups(channel)
                 # Is the channel now empty and needs deleting?
                 if queue.empty():
-                    del self.channels[channel]
+                    self.channels.pop(channel, None)
 
         # Group Expiration
         timeout = int(time.time()) - self.group_expiry
-        for group in self.groups:
-            for channel in list(self.groups.get(group, set())):
-                # If join time is older than group_expiry end the group membership
-                if (
-                    self.groups[group][channel]
-                    and int(self.groups[group][channel]) < timeout
-                ):
+        for channels in self.groups.values():
+
+            for name, timestamp in list(channels.items()):
```

Comment:
Why the extra `list()` call here?

Reply:
because of channels.pop(name, None). Otherwise the dict size changes and python fails